src/db/cassandra/rtmp_saver.py

```diff
@@ -0,0 +1,59 @@
+import librtmp
+import threading
+from datetime import datetime
+
+class RtmpSaver:
+  """Create a RTMP connection"""
+  def __init__(self, ip, port, stream_id, chunk_size=2097152):
+    self.stream_id = stream_id
+    self.connection_string = "rtmp://{0}:{1}/view/{2}".format(ip, port, stream_id)
+    self.connection = librtmp.RTMP(self.connection_string, live=True)
+    self.connection.connect()
+    self.stream = self.connection.create_stream()
+    self.chunk_size = chunk_size
+    self.bytes_read = 0
+    self.previous_read = 0
+    self.data = b''
+    self.is_reading = True
+
+  def start(self, file_service):
+    self.is_reading_lock = threading.Lock()
+    self.read_thread = threading.Thread(target=self._read, args=(file_service,))
+    self.read_thread.start()
+
+  def _read(self, file_service):
+    """Continuously read (chunk_size)MB chunks from RTMP server"""
+    self.is_reading_lock.acquire()
+    while self.is_reading:
+      self.is_reading_lock.release()
+      self.data += self.stream.read(self.chunk_size)
+      self.bytes_read = len(self.data)
+
+      if self.__is_last_chunk(self.bytes_read, self.previous_read):
```

Comment:
This method call has two underscores, while the method definition has 1.

Reply:
🤔